transcom/mymove
# B 17588 prime sim updates

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A843823&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

This story addresses updating the prime simulator to allow for functionality added in the above story. The solution required adding an additional `TextInput` that will allow the user to add a partial weight parameter to specific SIT service items being added in the payment request. If this input is left empty, it assumes full shipment weight.

![Screenshot 2023-11-22 at 8 51 53 AM](https://github.com/transcom/mymove/assets/136510600/61ccd320-e64e-4a4f-bbde-67b70575956f)
![Screenshot 2023-11-22 at 8 51 59 AM](https://github.com/transcom/mymove/assets/136510600/3b8ac605-e364-46f0-9c07-64abbfd4502f)

Also updated and added tests.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the MilMove app as the Prime
2. Find a move and click the `Create Payment Request` button at the top
3. Find SIT service items with the text input box
4. Either leave it empty or fill in a partial value
5. Log in as a TIO and find the move code and see the payment request with the values
